sql/src/main/java/org/apache/druid/sql/avatica/DruidConnection.java

```diff
@@ -97,7 +96,7 @@ public DruidJdbcStatement createStatement(SqlStatementFactory sqlStatementFactor
       final DruidJdbcStatement statement = new DruidJdbcStatement(
           connectionId,
           statementId,
-          context.copy(),
+          new HashMap<String, Object>(context),
```

Comment:
Can we move the copy into the ctor of `DruidJdbcStatement` so that no one forgets to make a copy?